docker.md

```diff
@@ -0,0 +1,102 @@
+# Docker
+
+- [Docker](#docker)
+  - [Build and host in Google Cloud Artifact Registry](#build-and-host-in-google-cloud-artifact-registry)
+    - [Step 1: Create AR Docker repository](#step-1-create-ar-docker-repository)
+    - [Step 2: Build and upload Docker image](#step-2-build-and-upload-docker-image)
+      - [Option A: Build using only Dockerfile](#option-a-build-using-only-dockerfile)
+      - [Option B: Build using Dockerfile and build config](#option-b-build-using-dockerfile-and-build-config)
+    - [Step 3: Confirm Docker image built and uploaded](#step-3-confirm-docker-image-built-and-uploaded)
+
+Docker is a tool designed to make it easier to create, deploy, and run
+applications. If you don't know what it is or how to use it, the
+[official docs](https://docs.docker.com/get-started/) are a good starting point.
+
+## Build and host in Google Cloud Artifact Registry
+
+You can host your Docker images in a Google Cloud
+[Artifact Registry](https://cloud.google.com/artifact-registry/docs/docker/quickstart)
+(AR) repository. Here are the basic steps for building a Docker image using
+[Cloud Build](https://cloud.google.com/build/docs/quickstart-build) and
+subsequently pushing it to AR - the following assume that you want to host it in
+the `australia-southeast1` region, and that you already have a `Dockerfile` in
+the working directory.
+
+### Step 1: Create AR Docker repository
+
+```shell
+LOCATION="australia-southeast1"
+PROJECT="<project>"
+REPO="<repo-name>"
+
+gcloud artifacts repositories create ${REPO} \
+    --repository-format "docker" \
+    --location ${LOCATION} \
+    --description "My awesome Docker repo" \
+    --project ${PROJECT}
+```
+
+Verify that it was created:
+
+```shell
+gcloud artifacts repositories list \
+    --project ${PROJECT} \
+    --location ${LOCATION}
+```
+
+### Step 2: Build and upload Docker image
+
+There are two ways to build Docker images with Cloud Build and upload to AR. For
+most cases, a `Dockerfile` is the only file required. For more complex build
+workflows, a build config is recommended. In both cases, the image is
+automatically uploaded to AR upon successful build.
+
+#### Option A: Build using only Dockerfile
+
+```shell
+IMG="<image-name>"
+TAG="<version-tag>"
+
+gcloud builds submit \
```

Comment:
Wondering if we should also include a `--timeout` here? It's certainly something I've been bitten by...